### PR TITLE
API review changes

### PR DESCRIPTION
**Pull request by ajcvickers**
- Various cleanup to RelationalTypeMapping classes
- Revert parameter mapping change and fix by changing DbType instead
- Re-order args on OpenAsync
- Small naming changes

### CODE REVIEW

**Comment by bricelam** [comment]
This seems to undo #8912 by @AndriySvyryd...

**Comment by ajcvickers** [comment]
It re-does it differently. All that was needed was that the DbType on the parameter is not "FixedLength". Discussed with @AndriySvyryd in person yesterday.

File: test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs [hunk 1]
```diff
@@ -74,7 +74,7 @@ public void Can_open_the_underlying_connection_async()
 
             context.Database.OpenConnectionAsync(cancellationToken);
 
-            connectionMock.Verify(m => m.OpenAsync(/*errorsExpected:*/ false, cancellationToken), Times.Once);
+            connectionMock.Verify(m => m.OpenAsync(cancellationToken, /*errorsExpected:*/ false), Times.Once);
```
Comment (smitpatel) [inline test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs]: Can uncomment the parameter name

File: test/EFCore.Relational.Tests/Update/BatchExecutorTest.cs [hunk 2]
```diff
@@ -58,7 +58,7 @@ public class BatchExecutorTest
 
             await batchExecutor.ExecuteAsync(new[] { mockModificationCommandBatch.Object }, mockRelationalConnection.Object, cancellationToken);
 
-            mockRelationalConnection.Verify(rc => rc.OpenAsync(/*errorsExpected:*/ false, cancellationToken));
+            mockRelationalConnection.Verify(rc => rc.OpenAsync(cancellationToken, /*errorsExpected:*/ false));
```
Comment (smitpatel) [inline test/EFCore.Relational.Tests/Update/BatchExecutorTest.cs]: Can uncomment the parameter name

File: src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs [hunk 3]
```diff
@@ -63,11 +63,11 @@ public virtual DbFunctionBuilder HasSchema([CanBeNull]string schema)
         ///     of expressions corresponding to the parameters passed to the function call. The callback should return a SQL fragment
         ///     expression representing the desired translation.
```
Comment (AndriySvyryd) [inline src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs]: The callback return doesn't have to be a SQL Expression anymore, perhaps add a links to the docs?

File: src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMapper.cs [hunk 4]
```diff
@@ -49,16 +49,16 @@ public class SqlServerTypeMapper : RelationalTypeMapper
...
-            = new SqlServerStringTypeMapping("nchar", dbType: DbType.StringFixedLength, unicode: true, size: null);
+            = new SqlServerStringTypeMapping("nchar", dbType: DbType.String, unicode: true);
 
         private readonly SqlServerStringTypeMapping _variableLengthUnicodeString
-            = new SqlServerStringTypeMapping("nvarchar", dbType: null, unicode: true, size: null);
+            = new SqlServerStringTypeMapping("nvarchar", dbType: null, unicode: true);
 
         private readonly SqlServerStringTypeMapping _fixedLengthAnsiString
-            = new SqlServerStringTypeMapping("char", dbType: DbType.AnsiStringFixedLength, unicode: false, size: null);
+            = new SqlServerStringTypeMapping("char", dbType: DbType.AnsiString);
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMapper.cs]: Wouldn't the make the column variable length in the schema? I thought we would only do this for parameters.

Comment (ajcvickers) [inline src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMapper.cs]: Nope. The only thing Migrations cares about is the store type name. DbTypes are only used for parameters.

File: src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs (hunk 3 above)
Comment (ajcvickers) [inline src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs]: @divega Do you want to create a fwlink for this?

File: src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMapper.cs (hunk 4 above)
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Storage/Internal/SqlServerTypeMapper.cs]: ADO.Net continues to confound me 😕

File: test/EFCore.Relational.Tests/Update/BatchExecutorTest.cs (hunk 2 above)
Comment (ajcvickers) [inline test/EFCore.Relational.Tests/Update/BatchExecutorTest.cs]: Both these are captured into an expression tree by MOQ, and expression trees can't have named arguments.

File: src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs (hunk 3 above)
Comment (divega) [inline src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs]: @ajcvickers https://go.microsoft.com/fwlink/?linkid=852477.

Comment (ajcvickers) [inline src/EFCore.Relational/Metadata/Builders/DbFunctionBuilder.cs]: Thanks @divega. Pushed it.

**Pull request merged**